satellite/metabase/rangedloop/service.go

```diff
@@ -50,15 +50,27 @@ func NewService(log *zap.Logger, config Config, provider RangeSplitter, observer
 // Improvement: track duration.
 type observerState struct {
 	observer       Observer
-	rangeObservers []Partial
+	rangeObservers []rangeObserverState
+}
+
+type rangeObserverState struct {
+	rangeObserver Partial
+	duration      *time.Duration
```

Comment:
Why make this a pointer?

Reply:
Sorry that it's taking so long.

This is a good point.

I'm passing rangeObserverState to the worker and then reading the duration afterwards. It seems to me if I want to do it like that I have to pass either `rangeObserverState` by reference or `rangeObserverState.duration`. And duration is the more granular of the two.

Ofcourse manipulating addresses from other threads is kind of an anti-pattern. It would be better to return by value from the goroutines. I'm not familiar enough with Go concurrency libraries that I know of an off-the-shelf solution for that. 

I could implement something myself with channels but I'm not sure if it's worth the effort in this case.

Reply:
I don't think we need to wrestle concurrency here. The range observer states are only created, modified, and then observed by one goroutine at a time, right?

I think you can update the duration using the index of the range observer in the slice:
```
for i, state := range states {
    ...
    states[i].duration += time.Since(start)
```

Alternatively, I don't see anything wrong with creating a []*rangeObserverState slice instead of []rangeObserverState in this instance.

Reply:
Changed `*duration` to `duration` and changed all `rangeObserverState` to `*rangeObserverState`.